### PR TITLE
Check for minimum password length upon self registration.

### LINKED ISSUE #2458: Self-registering allows weak password
### Description of the problem

When I self-register, I can use a password that consists of only 8 lowercase characters, while if I change/set a password from the jury interface (for the very same user I just created) it tells me that the password requires at least 10 characters.

### Your environment

DOMjudge `wfluxor-online` branch.

### Expected behaviour

I'd expect the same password policy to be applied.

### PR DESCRIPTION
**Pull request by nickygerritsen**
Fixes #2458.

### CODE REVIEW

File: webapp/src/Controller/SecurityController.php [hunk 1]
```diff
@@ -103,58 +105,62 @@ public function registerAction(
         $registration_form->handleRequest($request);
         if ($registration_form->isSubmitted() && $registration_form->isValid()) {
             $plainPass = $registration_form->get('plainPassword')->getData();
-            $password  = $passwordHasher->hashPassword($user, $plainPass);
-            $user->setPassword($password);
-            if ($user->getName() === null) {
-                $user->setName($user->getUsername());
-            }
+            if (strlen($plainPass) < UserController::MIN_PASSWORD_LENGTH) {
+                $this->addFlash('danger', "Password should be " . UserController::MIN_PASSWORD_LENGTH . "+ chars.");
```
Comment (vmcj) [inline webapp/src/Controller/SecurityController.php:109]:
```suggestion
                $this->addFlash('danger', "Password should be " . UserController::MIN_PASSWORD_LENGTH . "+ chars.");
                return $this->redirectToRoute('login');
```

Would make the PR so much easier to read I think.

All other changes are just indentations right?

File: webapp/src/Controller/SecurityController.php [hunk 2]
```diff
@@ -103,58 +105,62 @@ public function registerAction(
         $registration_form->handleRequest($request);
         if ($registration_form->isSubmitted() && $registration_form->isValid()) {
             $plainPass = $registration_form->get('plainPassword')->getData();
-            $password  = $passwordHasher->hashPassword($user, $plainPass);
-            $user->setPassword($password);
-            if ($user->getName() === null) {
-                $user->setName($user->getUsername());
-            }
+            if (strlen($plainPass) < UserController::MIN_PASSWORD_LENGTH) {
```
Comment (eldering) [inline webapp/src/Controller/SecurityController.php:108]: Wouldn't it make more sense to store this constant here in the security controller, or possibly in some other global place?

File: webapp/src/Controller/SecurityController.php (hunk 1 above)
Comment (eldering) [reply webapp/src/Controller/SecurityController.php:109]: All remaining changes are indentation indeed, and I agree that here it makes a lot of sense to return early, not just to make the diff simpler, but more importantly to reduce code nesting complexity.

Comment (nickygerritsen) [reply webapp/src/Controller/SecurityController.php:109]: We shouldnt return to login though, but register. will fix.

File: webapp/src/Controller/SecurityController.php (hunk 2 above)
Comment (nickygerritsen) [reply webapp/src/Controller/SecurityController.php:108]: Do you have a good suggestion? Since we also use it in the jury controller.

Comment (eldering) [reply webapp/src/Controller/SecurityController.php:108]: Maybe in `webapp/config/static.yaml.in`? Not sure whether we should also make it configurable from `configure` or so...

Comment (nickygerritsen) [reply webapp/src/Controller/SecurityController.php:108]: I do like that idea, but then maybe we do that in a separate issue / PR?

**Review by meisterT — APPROVED** [review]

**Comment by thijskh** [comment]
Probably we need to tell the user what length we expect when we present the form? Not reject it after submission with a surprise limit that was never communicated before?

**Comment by nickygerritsen** [comment]
> Probably we need to tell the user what length we expect when we present the form? Not reject it after submission with a surprise limit that was never communicated before?

The `minlength` html attribute kinda does this, right?

File: webapp/tests/Unit/Controller/PublicControllerTest.php [hunk 3]
```diff
@@ -210,17 +210,17 @@ public function selfRegisterProvider(): Generator
                     continue;
                 }
                 yield[['username'=>'minimaluser', 'teamName'=>'NewTeam','affiliation'=>'none'],'shirt-recognize-bar-together', $fixtures, $category];
-                yield[['username'=>'bruteforce', 'teamName'=>'Fib(4)','affiliation'=>'none'],'0112', $fixtures, $category];
-                yield[['username'=>'fullUser', 'name'=>'Full User', 'email'=>'email@domain.com','teamName'=>'Trial','affiliation'=>'none'],'.', $fixtures, $category];
+                yield[['username'=>'bruteforce', 'teamName'=>'Fib(4)','affiliation'=>'none'],'0112345678', $fixtures, $category];
```
Comment (vmcj) [inline webapp/tests/Unit/Controller/PublicControllerTest.php:213]:
```suggestion
                yield[['username'=>'bruteforce', 'teamName'=>'Fib(9)','affiliation'=>'none'],'01123581321', $fixtures, $category];
```

File: webapp/src/Controller/SecurityController.php (hunk 2 above)
Comment (nickygerritsen) [reply webapp/src/Controller/SecurityController.php]: I moved it to the same place as the enabling of removed intervals. Made sense to me.

**Comment by nickygerritsen** [comment]
> > Probably we need to tell the user what length we expect when we present the form? Not reject it after submission with a surprise limit that was never communicated before?
> 
> The `minlength` html attribute kinda does this, right?

I added an explicit help text for it.

**Pull request merged**